compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -258,7 +258,18 @@ impl<'a, 'b> BuildReducedGraphVisitor<'a, 'b> {
                 Ok(ty::Visibility::Restricted(DefId::local(CRATE_DEF_INDEX)))
             }
             ast::VisibilityKind::Inherited => {
-                Ok(ty::Visibility::Restricted(parent_scope.module.normal_ancestor_id))
+                if self.parent_scope.module.kind.is_enum() {
+                    // Anything in an enum has the visibility of the enum.
+                    Ok(*self
+                        .r
+                        .visibilities
+                        .get(&self.parent_scope.module.normal_ancestor_id.expect_local())
+                        .unwrap())
+                } else {
+                    // If it's not in an enum, its visibility is restricted to the trait or module
```

Comment:
```suggestion
                    // If it's not in an enum, the visibility is restricted to the current `mod` item.
```

Reply:
I think the way it is now is clearer.

Reply:
Now it's incorrect, `normal_ancestor_id` does not refer to traits and "module" is ambiguous in resolve context (it can be e.g. an unnamed block), while `normal_ancestor_id` refers to `mod` items specifically.

Reply:
Oh, I see what you mean.